File: bot.php
<?php
$botToken = "8581704909:AAET8NbghmL4eH7CMuglpLpdmCNDiHGzfe8";
$adminGroupId = "-1002123151908"; // Admin guruh ID sini o'zgartiring

$update = json_decode(file_get_contents('php://input'), true);

if (isset($update['message'])) {
    $message = $update['message'];
    $chatId = $message['chat']['id'];
    $text = $message['text'] ?? '';
    $userId = $message['from']['id'];
    $userName = $message['from']['first_name'] ?? 'Foydalanuvchi';
    
    // Foydalanuvchi ma'lumotlarini saqlash uchun
    $userFile = "users/{$userId}.json";
    
    if (!file_exists('users')) {
        mkdir('users', 0777, true);
    }
    
    $userData = [];
    if (file_exists($userFile)) {
        $userData = json_decode(file_get_contents($userFile), true);
    }
    
    // /start komandasi
    if ($text == '/start') {
        $userData = ['step' => 'waiting_destination'];
        file_put_contents($userFile, json_encode($userData));
        
        sendMessage($chatId, "🚖 *Taxi Bot*ga xush kelibsiz!\n\n🎯 Chiqish va borish manzilingizni yozing:\n\n_Masalan: Samarqand - Farg'ona va hokozo_");
    }
    
    // Borish joyi
    elseif ($userData['step'] == 'waiting_destination' && !empty($text)) {
        $userData['destination'] = $text;
        $userData['step'] = 'waiting_phone';
        file_put_contents($userFile, json_encode($userData));
        
        $keyboard = [
            'keyboard' => [
                [['text' => '📱 Telefon raqamni yuborish', 'request_contact' => true]]
            ],
            'resize_keyboard' => true,
            'one_time_keyboard' => true
        ];
        
        sendMessage($chatId, "✅ Borish joyi: *{$text}*\n\n📱 Telefon raqamingizni yuboring: masalan +998999558657", $keyboard);
    }
    
    // Telefon raqam qabul qilish (tugma orqali)
    elseif (isset($message['contact']) && $userData['step'] == 'waiting_phone') {
        $contact = $message['contact'];
        $phoneNumber = $contact['phone_number'];
        
        processOrder($chatId, $userId, $userName, $phoneNumber, $userData, $adminGroupId);
    }
    
    // Telefon raqam qabul qilish (qo'lda yozish)
    elseif ($userData['step'] == 'waiting_phone' && !empty($text)) {
        $phoneNumber = $text;
        
        processOrder($chatId, $userId, $userName, $phoneNumber, $userData, $adminGroupId);
    }
    
    else {
        sendMessage($chatId, "");
    }
}

function processOrder($chatId, $userId, $userName, $phoneNumber, &$userData, $adminGroupId) {
    $userData['phone'] = $phoneNumber;
    $userData['step'] = 'completed';
    file_put_contents("users/{$userId}.json", json_encode($userData));
    
    // Buyurtmani admin guruhga yuborish
    $orderText = "🚖 *YANGI TAXI BUYURTMA*\n\n";
    $orderText .= "👤 *Mijoz:* [" . $userName . "](tg://user?id=" . $userId . ")\n";
    $orderText .= "📱 *Telefon:* {$phoneNumber}\n";
    $orderText .= "🎯 *Qayerga:* {$userData['destination']}\n";
    $orderText .= "🕐 *Vaqt:* " . date('d.m.Y H:i') . "\n";
    $orderText .= "🆔 *User ID:* {$userId}";
    
    // Admin guruhga yuborish
    sendMessage($adminGroupId, $orderText);
    
    // Mijozga tasdiqlash
    $keyboard = [
        'keyboard' => [
            [['text' => '🚖 Yana taxi chaqirish']]
        ],
        'resize_keyboard' => true
    ];
    
    sendMessage($chatId, "✅ *Buyurtmangiz qabul qilindi!*\n\nTez orada haydovchi siz bilan bog'lanadi.\n\n📞 Telefon: {$phoneNumber}\n🎯 Qayerga: {$userData['destination']}", $keyboard);
}

function sendMessage($chatId, $text, $keyboard = null) {
    global $botToken;
    
    $data = [
        'chat_id' => $chatId,
        'text' => $text,
        'parse_mode' => 'Markdown'
    ];
    
    if ($keyboard) {
        $data['reply_markup'] = json_encode($keyboard);
    }
    
    $url = "https://api.telegram.org/bot{$botToken}/sendMessage";
    
    $ch = curl_init();
    curl_setopt($ch, CURLOPT_URL, $url);
    curl_setopt($ch, CURLOPT_POST, 1);
    curl_setopt($ch, CURLOPT_POSTFIELDS, http_build_query($data));
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
    
    $result = curl_exec($ch);
    curl_close($ch);
    
    return $result;
}
?>

File: bot.py
import asyncio
import json
import os
import re
from datetime import datetime
from dotenv import load_dotenv
from aiogram import Bot, Dispatcher, types, F
from aiogram.filters import Command
from aiogram.types import ReplyKeyboardMarkup, KeyboardButton, ReplyKeyboardRemove

load_dotenv()

BOT_TOKEN = os.environ["BOT_TOKEN"]
ADMIN_GROUP_ID = int(os.environ["ADMIN_GROUP_ID"])

bot = Bot(token=BOT_TOKEN)
dp = Dispatcher()

os.makedirs("users", exist_ok=True)


def safe_user_path(user_id: int) -> str:
    safe_id = re.sub(r"[^0-9]", "", str(user_id))
    path = os.path.realpath(f"users/{safe_id}.json")
    if not path.startswith(os.path.realpath("users")):
        raise ValueError("Invalid user_id")
    return path


def get_user(user_id: int) -> dict:
    path = safe_user_path(user_id)
    if os.path.exists(path):
        with open(path) as f:
            return json.load(f)
    return {}


def save_user(user_id: int, data: dict):
    with open(safe_user_path(user_id), "w") as f:
        json.dump(data, f)


@dp.message(Command("start"))
async def start(msg: types.Message):
    save_user(msg.from_user.id, {"step": "waiting_destination"})
    await msg.answer(
        "🚖 *Taxi Bot*ga xush kelibsiz!\n\n🎯 Chiqish va borish manzilingizni yozing:\n\n_Masalan: Samarqand - Farg'ona va hokozo_",
        parse_mode="Markdown",
        reply_markup=ReplyKeyboardRemove(),
    )


@dp.message(F.contact)
async def handle_contact(msg: types.Message):
    data = get_user(msg.from_user.id)
    if data.get("step") == "waiting_phone":
        await process_order(msg, data, msg.contact.phone_number)


@dp.message()
async def handle_text(msg: types.Message):
    user_id = msg.from_user.id
    text = msg.text or ""
    data = get_user(user_id)
    step = data.get("step")

    if text == "🚖 Yana taxi chaqirish":
        save_user(user_id, {"step": "waiting_destination"})
        await msg.answer("🎯 Chiqish va borish manzilingizni yozing:", reply_markup=ReplyKeyboardRemove())

    elif step == "waiting_destination" and text:
        data["destination"] = text
        data["step"] = "waiting_phone"
        save_user(user_id, data)

        kb = ReplyKeyboardMarkup(
            keyboard=[[KeyboardButton(text="📱 Telefon raqamni yuborish", request_contact=True)]],
            resize_keyboard=True,
            one_time_keyboard=True,
        )
        await msg.answer(
            f"✅ Borish joyi: *{text}*\n\n📱 Telefon raqamingizni yuboring: masalan +998999558657",
            parse_mode="Markdown",
            reply_markup=kb,
        )

    elif step == "waiting_phone" and text:
        await process_order(msg, data, text)


async def process_order(msg: types.Message, data: dict, phone: str):
    user_id = msg.from_user.id
    user_name = msg.from_user.first_name or "Foydalanuvchi"

    data["phone"] = phone
    data["step"] = "completed"
    save_user(user_id, data)

    order_text = (
        f"🚖 *YANGI TAXI BUYURTMA*\n\n"
        f"👤 *Mijoz:* [{user_name}](tg://user?id={user_id})\n"
        f"📱 *Telefon:* {phone}\n"
        f"🎯 *Qayerga:* {data['destination']}\n"
        f"🕐 *Vaqt:* {datetime.now().strftime('%d.%m.%Y %H:%M')}\n"
        f"🆔 *User ID:* {user_id}"
    )
    await bot.send_message(ADMIN_GROUP_ID, order_text, parse_mode="Markdown")

    kb = ReplyKeyboardMarkup(
        keyboard=[[KeyboardButton(text="🚖 Yana taxi chaqirish")]],
        resize_keyboard=True,
    )
    await msg.answer(
        f"✅ *Buyurtmangiz qabul qilindi!*\n\nTez orada haydovchi siz bilan bog'lanadi.\n\n📞 Telefon: {phone}\n🎯 Qayerga: {data['destination']}",
        parse_mode="Markdown",
        reply_markup=kb,
    )


async def main():
    await dp.start_polling(bot)


if __name__ == "__main__":
    asyncio.run(main())
